### ISSUE TITLE
(Graphics) GR COLORs are not accurate

### ISSUE THREAD

**Issue by Michaelangel007** [AppleWin/AppleWin#253]
The colors for the Lo-Resolution graphics screen range from OK to way off.

Running this AppleSoft Basic program on actual hardware and on AppleWin ...

```
100 GR
110 FOR X = 0 TO 39:FOR Y=0 TO 39
120 COLOR = INT(X/10) + 4 * (INT(Y/10))
130 PLOT X,Y:NEXT:NEXT
140 ? " 0=BLACK   1=RED     2=D.BLUE  3=PURPLE ";
150 ? " 4=D.GREEN 5=GRAY1   6=M.BLUE  7=L.BLUE ";
160 ? " 8=BROWN   9=ORANGE 10=GRAY2  11=PINK   ";
170 ? "12=L.GRN  13=YELLOW 14=AQUA   15=WHITE ";
180 GET A$:END
```

![apple_gr_level_correct](https://cloud.githubusercontent.com/assets/7876796/5442561/2333a57c-844f-11e4-84c5-3535c9bfd2e3.png)

![gr_applewin_1 25](https://cloud.githubusercontent.com/assets/7876796/5442582/42e99e8a-844f-11e4-9035-0c428327db9b.png)

**Comment by Michaelangel007** [comment]
Finally, a good example of GR "color bleed":

    FOR I=0 TO 15:FOR J=0 TO 15:COLOR=I:PLOT J*2,I:COLOR=J:PLOT J*2+1,I:NEXT:NEXT

**Comment by Michaelangel007** [comment]
Jace's NTSC color accuracy is surprisingly good:

![apple2_jace_gr](https://cloud.githubusercontent.com/assets/7876796/12253708/81954522-b896-11e5-9343-2710091b4778.png)

**Comment by sicklittlemonkey** [comment]
Yeah, Brendan has put a lot of work into the video colours and making video
cycle accurate.

It can run all the French Touch demos I think.

Cheers,
Nick.

**Comment by Michaelangel007** [comment]
Jace's NTSC GR reference image to compare with AppleWin and Sheldon's:

![apple2_jace_gr2](https://cloud.githubusercontent.com/assets/7876796/12283851/853441b4-b95f-11e5-9173-39e677efb539.png)

**Comment by Michaelangel007** [comment]
```
FOR I=0 TO 15:FOR J=0 TO 15:COLOR=I:PLOT J*2,I:COLOR=J:PLOT J*2+1,I:NEXT:NEXT
```

**Comment by Michaelangel007** [comment]
Sheldon's gray1 and gray2 are incorrect as they have chroma. Correct values are:

* gray1 = 0x838383, 0x787878, 0x838383, 0x787878
* gray2 = 0x787878, 0x838383, 0x787878, 0x838383

**Comment by Michaelangel007** [comment]
A decade later, an alternative version: ;-)

```BASIC
10 S=10:GR
20 FOR Y=0 TO 3:FOR X=0 TO 3:COLOR=Y*4+X:U=X*S:V=Y*S
30 FOR J=0 TO S-1:FOR I=0 TO S-1:PLOT U+I,V+J:NEXT:NEXT:NEXT:NEXT
```
